src/loops.rs

```diff
@@ -92,6 +97,35 @@ impl LintPass for LoopsPass {
                     }
                 }
             }
+
+            // Look for variables that are incremented once per loop iteration.
+            let mut visitor = IncrementVisitor { cx: cx, states: HashMap::new(), depth: 0, done: false };
+            walk_expr(&mut visitor, body);
+
+            // For each candidate, check the parent block to see if
+            // it's initialized to zero at the start of the loop.
+            let map = &cx.tcx.map;
+            let parent_scope = map.get_enclosing_scope(expr.id).and_then(|id| map.get_enclosing_scope(id) );
+            if let Some(parent_id) = parent_scope {
+                if let NodeBlock(block) = map.get(parent_id) {
+                    for (id, _) in visitor.states.iter().filter( |&(_,v)| *v == VarState::IncrOnce) {
+                        let mut visitor2 = InitializeVisitor { cx: cx, end_expr: expr, var_id: id.clone(),
+                                                               state: VarState::IncrOnce, name: None,
```

Comment:
This visitor is only looking at variables that we've already identified as being incremented exactly once in the loop body.